cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/service/subnetselector/SubnetSelectorStrategyMultiplePreferPublic.java

```diff
@@ -1,53 +1,39 @@
 package com.sequenceiq.cloudbreak.cloud.aws.service.subnetselector;
 
-import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Map;
 
 import javax.inject.Inject;
 
-import org.springframework.beans.factory.annotation.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
-import com.google.common.annotations.VisibleForTesting;
 import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
 import com.sequenceiq.cloudbreak.cloud.model.SubnetSelectionResult;
 
 @Component
 public class SubnetSelectorStrategyMultiplePreferPublic extends SubnetSelectorStrategy {
 
-    @VisibleForTesting
-    @Value("${cb.aws.subnet.different.az.min:2}")
-    int minSubnetCountInDifferentAz;
-
-    @VisibleForTesting
-    @Value("${cb.aws.subnet.different.az.max:3}")
-    int maxSubnetCountInDifferentAz;
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubnetSelectorStrategyMultiplePreferPublic.class);
 
     @Inject
     private SubnetSelectorService subnetSelectorService;
 
     @Override
-    protected SubnetSelectionResult selectInternal(List<CloudSubnet> subnetMetas) {
-        Map<String, CloudSubnet> selectedSubnets = subnetSelectorService.collectOnePublicSubnetPerAz(subnetMetas, maxSubnetCountInDifferentAz);
-        if (selectedSubnets.size() < minSubnetCountInDifferentAz) {
-            Map<String, CloudSubnet> privateSubnetsPerAz = subnetSelectorService.collectOnePrivateSubnetPerAz(subnetMetas, maxSubnetCountInDifferentAz);
-            subnetSelectorService.collectSubnetsOfMissingAz(selectedSubnets, privateSubnetsPerAz,
-                    minSubnetCountInDifferentAz);
-            if (selectedSubnets.size() < minSubnetCountInDifferentAz) {
-                return new SubnetSelectionResult(String.format(NOT_ENOUGH_AZ, selectedSubnets.size(), minSubnetCountInDifferentAz));
-            }
+    public SubnetSelectionResult select(Collection<CloudSubnet> subnets) {
+        List<CloudSubnet> result = subnetSelectorService.collectPublicSubnets(subnets);
+        LOGGER.info("Collected pubic subnets are: {}", result);
+        if (result.isEmpty()) {
```

Comment:
Consider the case we have 3 subnets:
AZ-a: public (mapPublicIpOnLaunch=true)
AZ-b: private
AZ-c: private

Now, the filter method would return AZ-a only. In this case the select step after the filter will throw an error because fallback private subnets are not present.

A similar problem would be with the PreferPrivate variant.